pkg/systemd/services/services.jsx

```diff
@@ -350,42 +413,26 @@
         return service_tabs_suffixes.includes(suffix);
     }
 
-    /* When the page is running in the background fetch only information about failed units
-     * in order to update the 'Page Status'. The whole listUnits is very expensive.
-     * We still need to maintain the 'unit_by_path' state object so that if we receive
-     * some signal we can normally parse it and update only the affected unit state
-     * instead of calling ListUnitsFiltered API call for every received signal which
-     * might have changed the failed units array
-     */
+    /* When the page is running in the background, fetch only information about failed units
+     * in order to update the 'Page Status'. */
     listFailedUnits() {
         return systemd_client[this.props.owner].call(s_bus.O_MANAGER, s_bus.I_MANAGER, "ListUnitsFiltered", [["failed"]])
                 .then(([failed]) => {
-                    failed.forEach(result => {
-                        const path = result[6];
-                        const unit_id = result[0];
-
-                        if (!this.isUnitHandled(unit_id))
+                    const units = {};
+                    failed.forEach(([
+                        Id, Description, LoadState, ActiveState, _substate, _followUnit, ObjectPath,
+                        _is_job_queued, _job_type, _job_path
+                    ]) => {
+                        if (!this.isUnitHandled(Id))
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230307-065433-1c06e6c8-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)